draft-lenders-dns-over-coap.md

```diff
@@ -265,6 +265,18 @@ the server.
 CoAP/CoRE Integration
 =====================
 
+DoC server considerations
+-------------------------
+In the case of CNAME records in a DNS response, a DoC server SHOULD follow common DNS resolver
+behavior {{?RFC1034}} by resolving a CNAME until the originally requested resource record type is
+reached. This reduces the number of message exchanges within an LLN.
+
+If not explicitly requested in the DNS query, it is RECOMMENDED to strip all record entries of type
+NS and class IN and all associated A and AAAA records from DNS responses received from upstream DNS
+resolvers.
+The NS record can not be used effectively by DoC anyway and tend to increase DNS response sizes
+unnecessarily.
+
```

Comment:
for the moment, maybe we should write:

The DoC server should try to send compact answers, i.e., add additional or authority sections in the DNS reply only if needed or anticipated that they help the DoC client to reduce additional queries.

Reply:
> The DoC server should send compact answers, i.e., add additional or authority sections in the DNS reply only if needed or anticipated that they help the DoC client to reduce additional queries.

Reworded slightly in abbef7f:

> The DoC server SHOULD send compact answers, i.e., add additional or authority sections in the DNS
> reply should only be sent if needed or anticipated that they help the DoC client to reduce
> additional queries. Any additional information MAY be stripped from the DNS response.

Reply:
the rewording includes mistakes (add additional ... reply should be sent"), and the second sentence blows up the text and doesn't give more information.

Reply:
> […] and the second sentence blows up the text and doesn't give more information.

Mh, don't you think it is important to give the reader explicit permission to strip a given response? Without that sentence, this might not be clear that this is an option.

Reply:
> the rewording includes mistakes (add additional ... reply should be sent"), 

Fixed

Reply:
no. as currently writte it probably will conflict with the DNS spec.

Reply:
Fine. Then I rather strip that sentence ;-) 427ffdd